### PR TITLE
Chapter 9 : Advanced Login

### PR DESCRIPTION
**Pull request by DaudKhadijah01**
ready

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -7,4 +7,32 @@ class User < ApplicationRecord
...
+
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+        BCrypt::Engine::MIN_COST
+      else
+        BCrypt::Engine.cost
+      end
+       BCrypt::Password.create string, cost: cost
+    end
+  end
```
Comment (ghost) [inline app/models/user.rb]: ident end with def in line 16

File: app/views/sessions/new.html.erb [hunk 2]
```diff
@@ -7,6 +7,11 @@
       <%= f.email_field :email, class: "form-control" %>
       <%= f.label :password %>
       <%= f.password_field :password, class: "form-control" %>
+      <%= f.label :remember_me, class: "checkbox inline" do %>
+          <%= f.check_box :remember_me %>
+          <span>Remember me on this computer</span>
+        <% end %>
```
Comment (ghost) [inline app/views/sessions/new.html.erb]: Wrong ident

**Review by ghost — COMMENTED** [review]
cmt

**Comment by DaudKhadijah01** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -16,7 +24,24 @@ def logged_in?
...
     @current_user = nil
   end
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def authenticated? remember_token
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: it shouldn't define here
please check again

**Comment by DaudKhadijah01** [comment]
ready

File: app/models/user.rb [hunk 4]
```diff
@@ -7,4 +7,32 @@ class User < ApplicationRecord
   validates :password, presence: true, length: {minimum: 6}
 
   has_secure_password
+
+    class << self
```
Comment (huongnt-2545) [inline app/models/user.rb]: indent

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
can i merge after this?

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -16,7 +24,20 @@ def logged_in?
...
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget(user)
+   user.forget
+   cookies.delete(:user_id)
+   cookies.delete(:remember_token)
```
Comment (ghost) [inline app/helpers/sessions_helper.rb]: use 2 space to indent not 1

**Review by ghost — COMMENTED** [review]
small cmt

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**